src/features/ProfilePage/components/PublicInfo.tsx

```diff
@@ -1,35 +1,81 @@
+import { useState } from 'react';
 import { useTranslation } from 'react-i18next';
 
-import { selectUserInfo } from '@/features/Authentication/userSlice';
+import { selectMentor, selectUser } from '@/features/Authentication/userSlice';
 import { useAppSelector } from '@/store';
-
-import { DEFAULT_ICON_SIZE, palette } from '@/components/constants';
+import { useDebounce } from '@/hooks/useDebounce';
+import { useUpdateMentorMutation } from '@/features/MentorPage/mentorPageApi';
+import { useUpdateUserMutation } from '@/features/Authentication/authenticationApi';
+
+import {
+  DEFAULT_ICON_SIZE,
+  palette,
+  SAVE_DELAY_MS,
+} from '@/components/constants';
 import LabeledInput from '@/components/LabeledInput';
 import Slider from '@/components/Slider';
 import Text from '@/components/Text';
 import TextInput from '@/components/TextInput';
 
+import type {
+  MentorUser,
+  User,
+} from '@/features/Authentication/authenticationApi';
+
 const PublicInfo = () => {
   const { t } = useTranslation('profile');
-  const userInfo = useAppSelector(selectUserInfo);
+  const user = useAppSelector(selectUser);
+  const mentor = useAppSelector(selectMentor);
+  const [updateUser] = useUpdateUserMutation();
+  const [updateMentor] = useUpdateMentorMutation();
+
+  const saveMentorData = useDebounce(
+    (mentor: MentorUser) => updateMentor(mentor),
+    SAVE_DELAY_MS,
+  );
+
+  const saveUserAndMentorData = useDebounce(
+    (mentor: MentorUser, user: User) => {
+      updateUser(user);
+      updateMentor(mentor);
+    },
+    SAVE_DELAY_MS,
+  );
+
+  const [localData, setLocalData] = useState<MentorUser>(mentor);
+  const [skillSearchValue, setSkillSearchValue] = useState('');
+
+  const updateMentorData = (key: keyof MentorUser, value: string) => {
+    setLocalData({ ...localData, [key]: value });
+    saveMentorData({ ...mentor, [key]: value });
+  };
+
+  const handleDisplayNameChange = (display_name: string) => {
+    setLocalData({ ...localData, display_name });
+    saveUserAndMentorData(
+      { ...mentor, display_name },
+      { ...user, display_name },
+    );
+  };
+
+  const handleBirthYearChange = (birth_year: string) => {
+    setLocalData({ ...localData, birth_year: Number(birth_year) });
+    saveMentorData({ ...mentor, birth_year: Number(birth_year) });
+  };
+
+  const handleIsVacationingChange = () => {
+    setLocalData({ ...localData, is_vacationing: !localData.is_vacationing });
+    saveMentorData({ ...mentor, is_vacationing: !localData.is_vacationing });
+  };
 
-  const [displayName, setDisplayName] = useState('');
-  const [birthYear, setBirthYear] = useState('');
-  const [area, setArea] = useState('');
-  const [status, setStatus] = useState('');
-  const [isActive, setIsActive] = useState(true);
-  const [story, setStory] = useState('');
-  const [topicSearchValue, setTopicSearchValue] = useState('');
+  const handleRegionChange = (value: string) =>
+    updateMentorData('region', value);
```

Comment:
Does this bring extra value, i would rather just inline the updateMentorData in input-onChange. One less fn created

Reply:
I think it's nicer, but it's just a preference. 🤷

Reply:
I can do it your way, it's more concise, true.